### ISSUE TITLE
operation error SSO OIDC: RegisterClient, failed to resolve service endpoint, an AWS region is required, but was not found

### ISSUE THREAD

**Issue by IskanderNovena** [common-fate/granted#519]
After upgrading from version 0.17.1 to 0.19.0, when using `assume` I get the following error:
`operation error SSO OIDC: RegisterClient, failed to resolve service endpoint, an AWS region is required, but was not found`

Nothing has changed in my settings, and I see no mention of any breaking change in the release notes of the versions after 0.17.1.

The `region` is present in the profile, and I've done an `aws sso login --profile <profilename>` before doing the `assume`.

**Comment by IskanderNovena** [comment]
Not sure, but this looks like the culprit: https://github.com/common-fate/granted/commit/e53483843490a7f289be34af66a5c785003ce0dc

**Comment by shwethaumashanker** [comment]
Thanks, @IskanderNovena, I am able to replicate the error. Currently woking on a fix, as a temporary solution please manually downgrade your Granted version: https://docs.commonfate.io/granted/getting-started#reverting-to-an-older-version-of-granted

**Comment by shwethaumashanker** [comment]
@IskanderNovena We just released [v0.19.1](https://github.com/common-fate/granted/releases/tag/v0.19.1) which should fix this issue. Please let me know if it works for you

**Comment by MDBeudekerCN** [comment]
Hi @shwethaumashanker 
It looks like the new release fixed the issue
Thanks for the quick action!

**Comment by IskanderNovena** [comment]
@shwethaumashanker I can confirm that version `0.19.2` is working for me (didn't have a chance to test `0.19.1`).
Thanks!

**Comment by shwethaumashanker** [comment]
@danielhanold, I tried replicating the issue with the following configuration that I used last time to replicate it, but it seems to be working for me. Can you please share the AWS profile that is causing this to fail for you? (Feel free to redact any account IDs).
```
[profile sso-profile]
sso_start_url = https://d-12345678.awsapps.com/start#/
sso_region = eu-west-1
sso_account_id = 123456789
sso_role_name = sso_role
region = eu-central-1
output = text

[profile demo]
role_arn = arn:aws:iam::123456789:role/Administrator
source_profile = sso-profile
region = us-west-2
output = json
```

**Comment by danielhanold** [comment]
@shwethaumashanker Thank you for the quick response.

For the setup of my AWS config, I followed this tutorial from AWS: https://docs.aws.amazon.com/cli/latest/userguide/sso-configure-profile-token.html. This resulted in the following configuration (which worked in version `0.19.2`):

```yaml
[sso-session xyz-sso]
sso_start_url = https://abc.awsapps.com/start
sso_region = us-west-2
sso_registration_scopes = sso:account:access

[profile Test_AWSAdministratorAccess]
sso_session = xyz-sso
sso_account_id = 12345678
sso_role_name = AWSAdministratorAccess
region = us-west-2
```

After you told me that this worked in your testing, I modified my configuration and found the following to work:
```yaml
[sso-session xyz-sso] # unused in this example
sso_start_url = https://abc.awsapps.com/start
sso_region = us-west-2
sso_registration_scopes = sso:account:access

[profile Test_AWSAdministratorAccess]
sso_start_url = https://abc.awsapps.com/start
sso_region = us-west-2
sso_account_id = 12345678
sso_role_name = AWSAdministratorAccess
region = us-west-2
```

It looks like version `0.20.0` is not compatible with the `sso-session` setup.

**Comment by danielhanold** [comment]
@JoshuaWilkes @shwethaumashanker Thanks very much for getting this fixed this quickly. Confirmed fixed in version `0.20.2`. You're exceeding any expectations for an already amazing tool 🚀
